src/components/common/SafeTokenWidget/index.tsx

```diff
@@ -2,31 +2,43 @@ import { SafeAppsTag, SAFE_TOKEN_ADDRESSES } from '@/config/constants'
 import { AppRoutes } from '@/config/routes'
 import { useRemoteSafeApps } from '@/hooks/safe-apps/useRemoteSafeApps'
 import useChainId from '@/hooks/useChainId'
-import useSafeTokenAllocation from '@/hooks/useSafeTokenAllocation'
+import useSafeTokenAllocation, { useSafeTokenBalance, type Vesting } from '@/hooks/useSafeTokenAllocation'
 import { OVERVIEW_EVENTS } from '@/services/analytics'
 import { formatVisualAmount } from '@/utils/formatters'
-import { Box, ButtonBase, Skeleton, Tooltip, Typography } from '@mui/material'
+import { Box, Button, ButtonBase, Skeleton, Tooltip, Typography } from '@mui/material'
 import { BigNumber } from 'ethers'
 import Link from 'next/link'
 import { useRouter } from 'next/router'
 import type { UrlObject } from 'url'
 import Track from '../Track'
 import SafeTokenIcon from '@/public/images/common/safe-token.svg'
 import css from './styles.module.css'
+import UnreadBadge from '../UnreadBadge'
 
 const TOKEN_DECIMALS = 18
 
 export const getSafeTokenAddress = (chainId: string): string => {
   return SAFE_TOKEN_ADDRESSES[chainId]
 }
 
+const canRedeemSep5Airdrop = (allocation?: Vesting[]): boolean => {
+  const sep5Allocation = allocation?.find(({ tag }) => tag === 'user_v2')
+
+  if (!sep5Allocation) {
+    return false
+  }
+
+  return !sep5Allocation.isRedeemed
```

Comment:
Like this we would need to make a change once the deadline is reached.
I think also checking if the allocation is expired would fix that.
```suggestion
  return !sep5Allocation.isRedeemed && !sep5Allocation.isExpired
```

Reply:
I've added the extra check in ba34f38bd794ba7d4de458ff6aca8551b5d101d4.